src/index.js

```diff
@@ -0,0 +1,176 @@
+    }
+
+    log(`ipns entry for ${value} is valid`)
+    return callback(null, null)
+  })
+}
+
+/**
+ * Validates the given ipns entry against the given public key.
+ *
+ * @param {Object} publicKey public key for validating the record.
+ * @param {Object} entry ipns entry record.
+ * @param {function(Error)} [callback]
+ * @returns {Promise|void}
+ */
+const embedPublicKey = (publicKey, entry, callback) => {
+  return callback(new Error('not implemented yet'))
+}
+
+/**
+ * Extracts a public key matching `pid` from the ipns record.
+ *
+ * @param {Object} peerId peer identifier object.
+ * @param {Object} entry ipns entry record.
+ * @param {function(Error)} [callback]
+ * @returns {Promise|void}
+ */
+const extractPublicKey = (peerId, entry, callback) => {
+  return callback(new Error('not implemented yet'))
+}
+
+// rawStdEncoding as go
+// TODO Remove once resolved
+// Created PR for allowing this inside base32-encode https://github.com/LinusU/base32-encode/issues/2
+const regex = new RegExp('=', 'g')
+const rawStdEncoding = (key) => base32Encode(key, 'RFC4648').replace(regex, '')
+
+/**
+ * Get key for storing the record in the datastore.
+ * Format: /ipns/${base32(<HASH>)}
+ *
+ * @param {Buffer} key peer identifier object.
+ * @returns {string}
+ */
+const getDatastoreKey = (key) => `/ipns/${rawStdEncoding(key)}`
+
+/**
+ * Get key for sharing the record in the routing mechanism.
+ * Format: ${base32(/ipns/<HASH>)}, ${base32(/pk/<HASH>)}
+ *
+ * @param {Buffer} key peer identifier object.
+ * @returns {string}
+ */
+const getIdKeys = (key) => {
+  const pkBuffer = Buffer.from('/pk/')
+  const ipnsBuffer = Buffer.from('/ipns/')
+
+  return {
+    nameKey: rawStdEncoding(Buffer.concat([pkBuffer, key])),
+    ipnsKey: rawStdEncoding(Buffer.concat([ipnsBuffer, key]))
+  }
+}
+
+// Sign ipns record data
+const sign = (privateKey, value, validityType, validity, callback) => {
+  const dataForSignature = ipnsEntryDataForSig(value, validityType, validity)
+
+  privateKey.sign(dataForSignature, (err, signature) => {
+    if (err) {
+      return callback(err)
+    }
+    return callback(null, signature)
+  })
+}
+
+// Create record data for being signed
+const ipnsEntryDataForSig = (value, validityType, eol) => {
+  const valueBuffer = Buffer.from(value)
+  const validityTypeBuffer = Buffer.from(validityType.toString())
+  const eolBuffer = Buffer.from(eol)
+
+  return Buffer.concat([valueBuffer, validityTypeBuffer, eolBuffer])
+}
+
+module.exports = {
+  // create ipns entry record
+  create,
+  // validate ipns entry record
+  validate,
+  // embed public key in the record
+  embedPublicKey,
+  // extract public key from the record
+  extractPublicKey,
+  // get key for datastore
+  getDatastoreKey,
+  // get keys for routing
+  getIdKeys,
+  // marshal
+  marshal: ipnsEntryProto.encode,
+  // unmarshal
+  unmarshal: ipnsEntryProto.decode
+}
```

Comment:
Most of these aren't in the API docs on the README yet? Is that on purpose?

Reply:
added now!